File: Paintroid/src/androidTest/java/org/catrobat/paintroid/test/junit/FileIOTest.java
package org.catrobat.paintroid.test.junit;

import android.graphics.Bitmap;

import org.catrobat.paintroid.FileIO;
import org.junit.Test;
import org.mockito.Mock;

import androidx.exifinterface.media.ExifInterface;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class FileIOTest {

	@Mock
	Bitmap bitmap;
	@Mock
	ExifInterface exifInterface;

	@Test
	public void testGetOrientedBitmap() {
		assertNull(FileIO.getOrientedBitmap(bitmap, 0));
	}

	@Test
	public void testGetBitmapOrientation() {
		assertEquals(0, FileIO.getBitmapOrientation(exifInterface), 0);
		exifInterface = mock(ExifInterface.class);
		when(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)).thenReturn(ExifInterface.ORIENTATION_ROTATE_90);
		assertEquals(90f, FileIO.getBitmapOrientation(exifInterface), 0);
		when(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)).thenReturn(ExifInterface.ORIENTATION_ROTATE_180);
		assertEquals(180f, FileIO.getBitmapOrientation(exifInterface), 0);
		when(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)).thenReturn(ExifInterface.ORIENTATION_ROTATE_270);
		assertEquals(270f, FileIO.getBitmapOrientation(exifInterface), 0);
		when(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)).thenReturn(ExifInterface.ORIENTATION_NORMAL);
		assertEquals(0f, FileIO.getBitmapOrientation(exifInterface), 0);
	}
}

File: Paintroid/src/androidTest/java/org/catrobat/paintroid/test/junit/FileIOTest.kt
package org.catrobat.paintroid.test.junit

import org.catrobat.paintroid.FileIO.getOrientedBitmap
import org.catrobat.paintroid.FileIO.getBitmapOrientation
import org.mockito.Mock
import android.graphics.Bitmap
import androidx.exifinterface.media.ExifInterface
import org.junit.Assert
import org.junit.Test
import org.mockito.Mockito

class FileIOTest {
    @Mock
    var bitmap: Bitmap? = null

    @Mock
    lateinit var exifInterface: ExifInterface

    @Test
    fun testGetOrientedBitmap() { Assert.assertNull(getOrientedBitmap(bitmap, 0f)) }

    @Test
    fun testGetBitmapOrientation() {
        Assert.assertEquals(0f, getBitmapOrientation(exifInterface), 0f)
        exifInterface = Mockito.mock(ExifInterface::class.java)
        Mockito.`when`(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
        ).thenReturn(ExifInterface.ORIENTATION_ROTATE_90)
        Assert.assertEquals(90f, getBitmapOrientation(exifInterface), 0f)
        Mockito.`when`(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
        ).thenReturn(ExifInterface.ORIENTATION_ROTATE_180)
        Assert.assertEquals(180f, getBitmapOrientation(exifInterface), 0f)
        Mockito.`when`(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
        ).thenReturn(ExifInterface.ORIENTATION_ROTATE_270)
        Assert.assertEquals(270f, getBitmapOrientation(exifInterface), 0f)
        Mockito.`when`(exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
        ).thenReturn(ExifInterface.ORIENTATION_NORMAL)
        Assert.assertEquals(0f, getBitmapOrientation(exifInterface), 0f)
    }
}
